docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Recommendation/OneClassMatrixFactorizationWithOptions.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.Trainers;
+
+namespace Samples.Dynamic.Trainers.Recommendation
+{
+    public static class OneClassMatrixFactorizationWithOptions
+    {
+        // This example shows the use of ML.NET's one-class matrix factorization module which implements
+        // Algorithm 1 in a <a href="https://www.csie.ntu.edu.tw/~cjlin/papers/one-class-mf/biased-mf-sdm-with-supp.pdf">paper</a>.
```

Comment:
>a [](start = 26, length = 1)

"Algorithm 1 in a paper" isn't very meaningful. Can you give a longer description of what it is?

Reply:
Now we tell user it's a coordinate descent method.


---
In reply to: [274574781](https://github.com/dotnet/machinelearning/pull/3282#discussion_r274574781) [](ancestors = 274574781)